core/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java

```diff
@@ -0,0 +1,313 @@
+        Row row = sheet.getRow(tableArea[0]);
+        List<String> headerNames = new ArrayList<>();
+        for (Cell cell : row) {
+            if (cell.getCellType() == CellType.STRING) {
+                headerNames.add(cell.getRichStringCellValue().getString());
+            } else {
+                break;
+            }
+        }
+        if (headerNames.size() == tableArea[3] - tableArea[2] + 1) {
+            tableArea[0]++;
+        } else {
+            headerNames.clear();
+            for (int col = tableArea[2]; col <= tableArea[3]; col++) {
+                headerNames.add("col" + col);
+            }
+        }
+        Table table = Table.create(options.tableName());
+        List<Column<?>> columns = new ArrayList<>(Collections.nCopies(headerNames.size(), null));
+        for (int rowNum = tableArea[0]; rowNum <= tableArea[1]; rowNum++) {
+            row = sheet.getRow(rowNum);
+            for (int colNum = 0; colNum < headerNames.size(); colNum++) {
+                Cell cell = row.getCell(colNum + tableArea[2], MissingCellPolicy.RETURN_BLANK_AS_NULL);
+                Column<?> column = columns.get(colNum);
+                if (cell != null) {
+                    if (column == null) {
+                        column = createColumn(headerNames.get(colNum), cell);
+                        columns.set(colNum, column);
+                        while (column.size() < rowNum - tableArea[0]) {
+                            column.appendMissing();
+                        }
+                    }
+                    Column<?> altColumn = appendValue(column, cell);
+                    if (altColumn != null && altColumn != column) {
+                        column = altColumn;
+                        columns.set(colNum, column);
+                    }
+                }
+                if (column != null) {
+                    while (column.size() <= rowNum - tableArea[0]) {
+                        column.appendMissing();
+                    }
+                }
+            }
+        }
+        columns.removeAll(Collections.singleton(null));
+        table.addColumns(columns.toArray(new Column<?>[columns.size()]));
+        return table;
+    }
+
+    private Column<?> appendValue(Column<?> column, Cell cell) {
+        switch (cell.getCellType()) {
+        case STRING:
+            column.appendCell(cell.getRichStringCellValue().getString());
+            return null;
+        case NUMERIC:
+            if (DateUtil.isCellDateFormatted(cell)) {
+                Date date = cell.getDateCellValue();
+                LocalDateTime localDate = LocalDateTime.of(date.getYear() + 1900, date.getMonth() + 1,
+                        date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds());
+                column.appendCell(localDate.toString());
+                return null;
+            } else {
+                double num = cell.getNumericCellValue();
+                if (column.type() == ColumnType.SHORT) {
```

Comment:
The other readers won't return a `Column<Short>`. I'd probably remove that type from here for consistency

https://github.com/jtablesaw/tablesaw/blob/f477a81bdee5e6309d59b5bcca19b66834d52ea1/core/src/main/java/tech/tablesaw/io/ColumnTypeDetector.java#L58

https://github.com/jtablesaw/tablesaw/commit/f477a81bdee5e6309d59b5bcca19b66834d52ea1